libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -539,12 +535,14 @@ internal class MapboxTripSession(
         tripService.updateNotification(progress)
         progress?.let {
             routeProgressObservers.forEach { it.onRouteProgressChanged(progress) }
-            checkBannerInstructionEvent(progress) { bannerInstruction ->
+            bannerInstructionEvent.isOccurring(progress)
+            checkBannerInstructionEvent { bannerInstruction ->
```

Comment:
Could you explain a little bit more what and why we're trying to fix?

As far as I understand, we don't anymore check if the banner instructions are in the progress, we will always append the latest.

Reply:
I've just updated the description. Let me know if it makes sense.

Reply:
Thanks! But now we're ignoring the result of `bannerInstructionEvent.isOccurring(progress)`, so we will push banner/voice instructions with each progress update, won't we?

If there are no new updates, we will push the old ones again. Or am I missing something?

Reply:
Great catch! Addressed in the latest commit pushed.

Reply:
If I'm not mistaken, we still have the same issue, don't we?

Let's imagine this scenario:
1. Route progress with the banner is generated (no observer, developer is not notified).
2. Developer registers observer and is notified.
3. Route progress **without** banner is generated (observer is already registered and the developer is notified again with the old value, `bannerInstructionEvent.latestBannerInstructions` is not null).

We should tie the `latestBannerInstructions` only to the moment when the developer registers the listener.

Reply:
We're actually doing that 👀 

https://github.com/mapbox/mapbox-navigation-android/blob/3ddd3c64f55988d0b6f9a8b8a0b9ccb4c57cc960/libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt#L333

https://github.com/mapbox/mapbox-navigation-android/blob/3ddd3c64f55988d0b6f9a8b8a0b9ccb4c57cc960/libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt#L557-L563

Same thing for `voiceInstructions`. Am I missing something?

Reply:
Okay this is resolved.